grid.py

```diff
@@ -69,13 +69,39 @@ def display_grid(self):
 
     def populated_with_words(self, iterable_keys, alt_index = 0, across_index = 0, down_index = 0):
         """
-        REMEMBER TO POP USED WORDS IN A WAY THAT DOESN'T SCREW WITH THE ITERATION THROUGH WORDS.
-        MAYBE PASS A COPY OF WORDLIST TO THE RECURSIVE FUNCTION WITH THE CURRENT WORD REMOVED?
+        Attempts to recursively populate the crossword grid with words from the wordlist.
+        This method uses a backtracking algorithm to fill the crossword grid by alternating 
+        between 'across' and 'down' words. It selects words from the provided wordlists that 
+        fit into the current grid configuration, ensuring that all intersecting words are 
+        valid. If a word placement leads to a dead end (i.e., no valid subsequent placements), 
+        the method backtracks and tries alternative words. The function returns a boolean based
+        on whether population is successful.
         """
-        # Stop populating with words if all across and down indexes have been iterated through
         if across_index == len(iterable_keys["across"]) and down_index == len(iterable_keys["down"]):
             return True
-        
+        direction, word_num, across_index, down_index = self.alternate_index_directions(across_index, down_index, alt_index, iterable_keys)
+        word_length = self.words[direction][word_num].length
+        start_y, start_x = self.words[direction][word_num].start_pos
+        current_word = [self._grid[start_y][start_x + i].letter if direction == "across" else self._grid[start_y + i][start_x].letter for i in range(word_length)]
```

Comment:
```suggestion
        # Check direction to determine whether to read across (horizontally) or down (vertically).
        current_word = [self._grid[start_y][start_x + i].letter if direction == "across" else self._grid[start_y + i][start_x].letter for i in range(word_length)]
```

An example of commenting description before or beside the code, this can inform collaborators what a particular line of code does. Very helpful in for the other devs when trying to figure out how the process of the function